src/vscode-bicep/src/webviews/draggable-resources-view/provider.ts

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as vscode from "vscode";
+import crypto from "crypto";
+
+export class DraggableResourcesViewProvider
+  implements vscode.WebviewViewProvider
+{
+  private readonly extensionUri: vscode.Uri;
+  private readonly rootUri: vscode.Uri;
+
+  constructor(extensionUri: vscode.Uri) {
+    this.extensionUri = vscode.Uri.joinPath(extensionUri, "..", "vscode-bicep-ui");
+    this.rootUri = vscode.Uri.joinPath(
+      this.extensionUri,
+      "apps",
+      "resource-type-explorer",
+      "dist",
+    );
+  }
+
+  resolveWebviewView(webviewView: vscode.WebviewView) {
+    webviewView.webview.options = {
+      enableScripts: true,
+      localResourceRoots: [this.rootUri],
+    };
+    webviewView.webview.html = this.createWebviewHtml(webviewView.webview);
+  }
+
+  private getWebviewResourceUri(
+    webview: vscode.Webview,
+    ...resourceName: string[]
+  ) {
+    return webview.asWebviewUri(
+      vscode.Uri.joinPath(this.rootUri, ...resourceName),
+    );
+  }
+
+  private createWebviewHtml(webview: vscode.Webview): string {
+    const { cspSource } = webview;
+    const nonce = crypto.randomBytes(16).toString("hex");
+    const stylesUri = this.getWebviewResourceUri(
+      webview,
+      "assets",
+      "index.css",
+    );
+    const scriptUri = this.getWebviewResourceUri(webview, "index.js");
+
+    const config = vscode.workspace.getConfiguration('workbench.colorCustomizations');
+    const editorBackground = config['editor.background'] || '#1e1e1e';
+    const editorForeground = config['editor.foreground'] || '#d4d4d4';
+    const listHoverBackground = config['list.hoverBackground'] || '#3c3c3c';
+
+    return `
+      <!DOCTYPE html>
+      <html lang="en">
+      <head>
+        <meta charset="UTF-8">
+        <!--
+        Use a content security policy to only allow loading images from our extension directory,
+        and only allow scripts that have a specific nonce.
+        -->
+        <meta http-equiv="Content-Security-Policy" content="default-src 'none'; font-src ${cspSource} data:; style-src ${cspSource} 'unsafe-inline'; img-src ${cspSource} data:; script-src 'unsafe-inline' vscode-webview-resource:;">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <link rel="stylesheet" type="text/css" href="${stylesUri}">
+        <style>
+          :root {
+              --vscode-editor-background: ${editorBackground};
+              --vscode-editor-foreground: ${editorForeground};
+              --vscode-list-hoverBackground: ${listHoverBackground};
```

Comment:
If I’m understanding correctly, the CSS variables don't need to be manually defined; they will be automatically injected by VS Code.

Reply:
Good call out, thanks. Removed here and updated to get sideBar variables instead of editor.